test/onnx/test_op_consistancy.py

```diff
@@ -0,0 +1,365 @@
+        )
+
+    def _contains_dtype(self, dtype: torch.dtype) -> bool:
+        return self.dtypes is None or dtype in self.dtypes
+
+    def should_fail(self, opset: int, dtype: torch.dtype) -> bool:
+        """Returns whether the test should fail for the given opset and dtype."""
+        return self._contains_opset(opset) and self._contains_dtype(dtype)
+
+
+def skip_ops(
+    all_opinfos: Sequence[opinfo_core.OpInfo],
+    test_case_name: str,
+    base_test_name: str,
+    to_skip: Iterable[DecorateMeta],
+):
+    """Decorates OpInfo tests with decorators based on the to_skip list."""
+    ops_mapping = {(info.name, info.variant_test_name): info for info in all_opinfos}
+    for decorate_meta in to_skip:
+        opinfo = ops_mapping.get((decorate_meta.op_name, decorate_meta.variant_name))
+        assert (
+            opinfo is not None
+        ), f"Couldn't find OpInfo for {decorate_meta}. Did you need to specify variant_name?"
+        decorators = list(opinfo.decorators)
+        new_decorator = opinfo_core.DecorateInfo(
+            decorate_meta.decorator,
+            test_case_name,
+            base_test_name,
+            device_type=decorate_meta.device_type,
+            dtypes=decorate_meta.dtypes,
+        )
+        decorators.append(new_decorator)
+        opinfo.decorators = tuple(decorators)
+
+    # This decorator doesn't modify fn in any way
+    def wrapped(fn):
+        return fn
+
+    return wrapped
+
+
+def opsets_before(opset: int) -> Callable[[int], bool]:
+    """Returns a comparison function that decides if the given opset is before the specified."""
+
+    def compare(other_opset: int):
+        return other_opset < opset
+
+    return compare
+
+
+def opsets_after(opset: int) -> Callable[[int], bool]:
+    """Returns a comparison function that decides if the given opset is after the specified."""
+
+    def compare(other_opset: int):
+        return other_opset > opset
+
+    return compare
+
+
+# Modify this section ###
+# NOTE: Modify this section as more ops are supported. The list should be sorted
+# alphabetically.
+
+# Ops to be tested for consistency between onnx and pytorch
+ALLOWLIST_OP = (
+    "ceil",
+    "sqrt",
+    "t",
+)
+
+# fmt: off
+# Turn off black formatting to keep the list compact
+
+# Expected failures for onnx export. If an op is expected to fail only for certain
+# ONNX opsets, add the op to EXPECTED_OPSET_FAILS below.
+# The list should be sorted alphabetically by op name.
+EXPECTED_SKIPS_OR_FAILS: Tuple[DecorateMeta, ...] = (
+    skip("ceil", dtypes=BOOL_TYPES + INT_TYPES + QINT_TYPES + COMPLEX_TYPES, reason="not supported by onnx"),
+    skip("ceil", dtypes=[torch.float64], reason="Ceil on f64 not supported by ONNX Runtime"),
+    skip("sqrt", dtypes=BOOL_TYPES + QINT_TYPES + COMPLEX_TYPES, reason="not supported by onnx"),
+    xfail("t", dtypes=COMPLEX_TYPES, reason="jit tracer error for complex types"),
+    skip("t", dtypes=[torch.float16], reason="flaky tests in CI"),
```

Comment:
Is this a precision issue? Should we add TODO to fix it with optional rtol/atol

Reply:
not sure; could be the non-held error that's just fixed. I will remove it and see what happens

Reply:
Removed